### ISSUE TITLE
Scanner Alert - It does not always alert.

### ISSUE THREAD

**Issue by vincentkenny01** [Der-Henning/tgtg#418]
I don't know if this is built-in phenomena but this has happened on a few occasions. 

Scanner sleep window is 29 seconds. 

Today, a surprise bag from my favourite store came online as they generally drop them at the same time everyday weekdays. I was on the app and I managed to grab the bag and paid for it but the scanner did not notify me at all that a bag was available. 
My question:
Is there a minimum time threshold that a bag is available for before the scanner alerts you? 

This has happened before as well where the bag is dropped and is sold within seconds but scanner does not alert. 

I am using it under docker with telegram. 

Just try to understand if there is a limit for how long the bag should be available for before it alerts you.

**Comment by vincentkenny01** [comment]
Also does anyone know what's the lowest interval you can use without getting blocked. I went down to 5seconds and got banned for 24hours. But 60 second sleep is too long for bags I want which disappear within seconds and there is no alert.

**Comment by lars-devs** [comment]
There’s no minimum threshold that triggers notification; more precisely, the scanner notifies You at any occasion when the items available count rises from zero to more than zero. This can be due to another user’s cancellation or regular daily increase. But still, the scanner recognizes any change - You can see that in Your output, while You won’t get notified about changes from e.g. 3 items to 2 items, as well as changes from 1 item to 2 items (due to cancellation), since there’s still at least one item available.

From my observations over the last couple of months, there’s no recognizable pattern according to which a TGTG scanner will be banned. Sometimes my scanner ran for two weeks without rate limit, sometimes just a few days. Same goes for the time it takes to unban. Just 29 seconds seems to be way too low in my opinion, since it is more likely to trigger the rate limit than my 10 minutes frequency. Sorry to hear that Your desired bag is sold so fast.

In conclusion, possible reasons for this behaviour might be:
- Possible rate limit due to low frequency (sometimes the unban happened without need to restart for me. You might not notice the rate limit at all)
- Bag was sold just seconds before the scanner scanned availablity, or
- You bought the item in Your app seconds before the scanner could see it.

**Comment by vincentkenny01** [comment]
Hi,
Thanks for a detailed response.
I think the bag gets sold within seconds of release like less 15 seconds and it’s gone. As there’s only one bag a day it makes it slightly more difficult. I’ve noticed this as I have managed to buy it within 10 seconds of release and scanner didn’t alert each time.
I’m running your program in docker environment and changed the sleep variable to 29 which I presume is 29 seconds.
How can I change the scan interval from docker as I don’t get to see other variables such as scan interval which you mentioned as 10min as default?
Best,
V

On Oct 30, 2023, at 01:30, lars-devs ***@***.***> wrote:

﻿

There’s no minimum threshold that triggers notification; more precisely, the scanner notifies You at any occasion when the items available count rises from zero to more than zero. This can be due to another user’s cancellation or regular daily increase. But still, the scanner recognizes any change - You can see that in Your output, while You won’t get notified about changes from e.g. 3 items to 2 items, as well as changes from 1 item to 2 items (due to cancellation), since there’s still at least one item available.

From my observations over the last couple of months, there’s no recognizable pattern according to which a TGTG scanner will be banned. Sometimes my scanner ran for two weeks without rate limit, sometimes just a few days. Same goes for the time it takes to unban. Just 29 seconds seems to be way too low in my opinion, since it is more likely to trigger the rate limit than my 10 minutes frequency. Sorry to hear that Your desired bag is sold so fast.

In conclusion, possible reasons for this behaviour might be:

  *   Possible rate limit due to low frequency (sometimes the unban happened without need to restart for me. You might not notice the rate limit at all)
  *   Bag was sold just seconds before the scanner scanned availablity, or
  *   You bought the item in Your app seconds before the scanner could see it.

**Comment by lars-devs** [comment]
You're right, using `SleepTime` sets the time in seconds, see `config.sample.ini`:

```
## Time to wait till next scan in seconds - default 60 seconds
## The Scanner will randomize the wait time by +/- 10%
```

10 minutes is the value I personally use and suits for me. Fortunately, I don't watch magic bags that get sold so fast.

Btw, I'm not the project's owner, I just contribute ^^.

**Comment by oryinchan** [comment]
How's 29 sec performance, do u still get ban fast?

**Comment by vincentkenny01** [comment]
I was getting ban every 2 to 3 weeks for few hours so changed it to 1 minute interval.

On Feb 20, 2024, at 23:47, oryinchan ***@***.***> wrote:

﻿

How's 29 sec performance, do u still get ban fast?
